scipy/integrate/meson.build

```diff
@@ -0,0 +1,87 @@
+fortran_ignore_warnings = [
+  '-Wno-tabs', '-Wno-conversion',
+  '-Wno-argument-mismatch', '-Wno-unused-dummy-argument',
+  '-Wno-maybe-uninitialized', '-Wno-unused-label',
+  '-Wno-unused-variable']
+
+mach_src = ['mach/d1mach.f', 'mach/xerror.f']
+quadpack_src = ['quadpack/dqagie.f', 
+    'quadpack/dqk15i.f',
+    'quadpack/dqawo.f',
+    'quadpack/dqelg.f',
+    'quadpack/dqage.f',
+    'quadpack/dqng.f', 
+    'quadpack/dqwgts.f',
+    'quadpack/dqwgtc.f',
+    'quadpack/dqag.f',
+    'quadpack/dqcheb.f',
+    'quadpack/dqc25s.f',
+    'quadpack/dqagpe.f',
+    'quadpack/dqk51.f',
+    'quadpack/dqk41.f',
+    'quadpack/dqagi.f',
+    'quadpack/dqags.f',
+    'quadpack/dqpsrt.f',
+    'quadpack/dqk15.f',
+    'quadpack/dqaws.f',
+    'quadpack/dqc25f.f',
+    'quadpack/dqwgtf.f',
+    'quadpack/dqawfe.f',
+    'quadpack/dqagse.f',
+    'quadpack/dqawf.f',
+    'quadpack/dqawse.f',
+    'quadpack/dqawc.f',
+    'quadpack/dqk61.f',
+    'quadpack/dqawoe.f',
+    'quadpack/dqagp.f',
+    'quadpack/dqc25c.f',
+    'quadpack/dqk15w.f',
+    'quadpack/dqk31.f',
+    'quadpack/dqmomo.f',
+    'quadpack/dqawce.f',
+    'quadpack/dqk21.f'
+]
+lsoda_src = ['odepack/blkdta000.f', 
+    'odepack/bnorm.f',
+    'odepack/cfode.f',
+    'odepack/ewset.f',
+    'odepack/fnorm.f',
+    'odepack/intdy.f',
+    'odepack/lsoda.f',
+    'odepack/prja.f',
+    'odepack/solsy.f',
+    'odepack/srcma.f',
+    'odepack/stoda.f',
+    'odepack/vmnorm.f',
+    'odepack/xerrwv.f',
+    'odepack/xsetf.f',
+    'odepack/xsetun.f'
+]
+vode_src = ['odepack/vode.f', 'odepack/zvode.f']
+dop_src = ['dop/dop853.f', 'dop/dopri5.f']
+quadpack_test_src = ['tests/_test_multivariate.c']
+odeint_banded_test_src = ['tests/banded5x5.f']
+
+mach = static_library('mach', 
+    mach_src,
+    fortran_args: fortran_ignore_warnings)
+quadpack = static_library('quadpack', 
+    quadpack_src,
+    fortran_args: fortran_ignore_warnings)
+lsoda = static_library('lsoda', 
+    lsoda_src,
+    fortran_args: fortran_ignore_warnings)
+vode = static_library('vode', 
+    vode_src,
+    fortran_args: fortran_ignore_warnings)
+dop = static_library('dop', 
+    dop_src,
+    fortran_args: fortran_ignore_warnings)
+
+py3.extension_module('_quadpack',
+    ['__quadpack.h', '_quadpackmodule.c'] + quadpack_src + mach_src,
+    fortran_args: fortran_ignore_warnings,
```

Comment:
When I add, `c_arg: numpy_noder_api` here, the build fails with something like following errors,
```cpp
.
.
.
../scipy/scipy/integrate/__quadpack.h:895:29: error: 'PyArrayObject' {aka 'struct tagPyArrayObject'} has no member named 'dat'
  895 |   rlist = (double *)ap_rlist->data;
      |                             ^~
../scipy/scipy/integrate/__quadpack.h:896:29: error: 'PyArrayObject' {aka 'struct tagPyArrayObject'} has no member named 'dat'
  896 |   elist = (double *)ap_elist->data;
      |                             ^~
../scipy/scipy/integrate/__quadpack.h: In function 'quadpack_qawse':
../scipy/scipy/integrate/__quadpack.h:975:26: error: 'PyArrayObject' {aka 'struct tagPyArrayObject'} has no member named 'dat'
  975 |   iord = (F_INT *)ap_iord->data;
.
.
.
```

Reply:
Any ideas why this might be happening?

Reply:
No idea, let me have a closer look.

Reply:
Ah yes, I do. Direct field access like `->data` is part of the old NumPy C API that is now deprecated. If you replace it with `PyArray_DATA` it will work.

The reason for this deprecation is to make the layout of the numpy ndarray object more opaque, so that it will become possible to make changes in the future. Unfortunately that has been going slow, because there's so much old code out there.

Reply:
There's quite a lot of lines of `->data`, but the replacement is quite mechanical. So I suggest to fix this in a separate PR to SciPy, and then we can rebase on top of that once it's merged. In the meantime, maybe just comment out the flag:
```
#c_args: numpy_nodepr_api,   # uncomment once PR-xxxx is merged
```

Reply:
For some extensions it may be too much work to replace all the deprecated API, this is bit of a judgement call. If so, just make a comment that this needs to be fixed in the future and leave it alone.

Reply:
I have opened a PR in main scipy repo. Resolving this for now.